src/Uri.php

```diff
@@ -53,7 +53,7 @@ class Uri implements UriInterface, Stringable
     public const CHAR_UNRESERVED = 'a-zA-Z0-9_\-\.~\pL';
 
     /** @var int[] Array indexed by valid scheme names to their corresponding ports. */
-    protected $allowedSchemes = [
+    protected array $allowedSchemes = [
```

Comment:
Adding types to protected members is a BC break: https://3v4l.org/frBrF

The documented type could be improved with `/** @var array<string, positive-int> */`